gcc/cp/cp-tree.h

```diff
@@ -3204,6 +3204,18 @@ struct GTY(()) lang_decl {
 #define CONTRACT_HELPER(NODE) \
  (LANG_DECL_FN_CHECK (NODE)->contract_helper)
 
+/* In VIEW_CONVERT_EXPR, set when this node is a const wrapper.  Used to
+   constify entities inside contract assertions.  */
+
+#define EXPR_CONTRACT_CONST_WRAPPER_P(NODE) \
+  (TREE_CHECK(NODE, VIEW_CONVERT_EXPR)->base.private_flag)
+
+/* Remove any VIEW_CONVERT_EXPR that's used to constify an entity inside a
+   contract assertion.  */
+
+#define STRIP_ANY_CONTRACT_CONST_WRAPPER(EXP) \
```

Comment:
Why do we need a macro instead of just calling the function where we need to use this?

Reply:
In other words, have the function take the tree by reference, modify the tree in the function, nuke the macro?

Reply:
Or just have the function return the possibly-different tree, and then <see below in the place where it's used..>

Reply:
done, renamed to strip_contract_constify_expr